app/lib/validation-helpers.js

```diff
@@ -183,10 +187,38 @@ const validateDob = (date, year) => {
   return null
 }
 
+const validateBreedForCountry = (value, helpers) => {
+  const elemName = 'breed'
+  const country = helpers.state.ancestors[0].country
+
+  if (value === 'XL Bully' && country === 'Scotland') {
+    return helpers.message(invalidBreedForCountryMessage, { path: [elemName] })
+  }
+
+  return value
+}
+
+const validateBreedForCountryChoosingAddress = async (personReference, payload, fieldName = 'address') => {
+  const ownerAndDogs = await getPersonAndDogs(personReference)
+
+  return (ownerAndDogs.dogs.some(dog => dog.breed === 'XL Bully') && payload.address.country === 'Scotland')
+    ? new Joi.ValidationError(invalidBreedForCountryMessage, [{ message: invalidBreedForCountryMessage, path: [fieldName], type: 'custom' }])
```

Comment:
does this give you the same Joi errors that we were trying to get with the code in the endpoints?  It's way nicer. 👍

Reply:
Yes :)